DHFPROD-2135: FE Implementation Display MLCP command in ingest step

xnikhil08 opened the pull request:
1. Updated the MLCP command with suggested changes related to step number and CSS.
2. Rebased the repository to resolve conflicts.

@wooldridge - Please review.

wooldridge commented on web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts:
```diff
@@ -1,16 +1,18 @@
-import { Component, OnInit, OnDestroy, ViewChild } from '@angular/core';
```
The ViewChild was added as part of the following: https://github.com/marklogic/marklogic-data-hub/pull/2689
I don't think it should be removed.

wooldridge commented on web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts:
```diff
@@ -33,18 +33,17 @@ import * as _ from "lodash";
 })
 export class EditFlowComponent implements OnInit, OnDestroy {
-
-  @ViewChild(EditFlowUiComponent) editFlowUi: EditFlowUiComponent;
```
The ViewChild code was added as part of the following: https://github.com/marklogic/marklogic-data-hub/pull/2689
I don't think it should be removed.

wooldridge commented on web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts:
```diff
@@ -180,7 +174,6 @@ export class EditFlowComponent implements OnInit, OnDestroy {
   }
   updateStep(step) {
     this.manageFlowsService.updateStep(this.flow.id, step.id, step).subscribe(resp => {
-      this.editFlowUi.stepUpdated(step);
```
The stepUpdated() was added as part of the following: https://github.com/marklogic/marklogic-data-hub/pull/2689
I don't think it should be removed.

wooldridge commented on web/src/main/ui/app/components/flows-new/edit-flow/edit-flow.component.ts:
```diff
@@ -23,8 +25,6 @@ import * as _ from "lodash";
     [selectedStepId]="selectedStepId"
     [projectDirectory]="projectDirectory"
     [flowEnded]="flowEnded"
-    [runFlowClicked]="runFlowClicked"
```
@brucean52 was this (and maybe other changes) part of DHFPROD-2627?

wooldridge commented on web/src/main/ui/app/components/flows-new/edit-flow/ingest/ingest.component.ts:
```diff
@@ -37,7 +47,6 @@ export class IngestComponent implements OnInit {
     } = this.step.fileLocations;
 
     const {
-      additionalCollections,
```
@brucean52 was this part of DHFPROD-2561? Want to make sure working code isn't deleted with this PR.

aebadirad commented:
This branch needs a rebase with develop. Could we get a rebase?

xnikhil08 commented:
I rebased the this branch with develop and implemented all the suggested changes.

@aebadirad & @wooldridge - Please review now and let me know if you still find any issues.

xnikhil08 commented:
> I'm testing this feature with the URI replacement information in the documentation here:
> 
> https://docs.marklogic.com/guide/mlcp/import
> 
> under "Transforming the Default URI".
> 
> When I type the following in the Target URI Replacement field:
> 
> `/space,'',/bill/data/,'/will/'`
> 
> I see this in the MLCP command:
> 
> `-output_uri_replace "/space,'',/bill/data/,'/will',''"`
> 
> It appears there is an extra '' adde to the comma-separated list. According to the documentation it should look like this:
> 
> `-output_uri_replace "/space,'',/bill/data/,'/will/'"`

@wooldridge - It is fixed. Please review now.

xnikhil08 commented:
> I'm testing the step index information that appears in the command. When I have a single Ingestion step, it looks good, we have:
> 
> step=0
> 
> But when I add additional Ingestion steps an select those, the command shows the following:
> 
> step=-1
> 
> See my screencast:
> 
> https://drive.google.com/file/d/1tb0xkRDywo56lacsVbC-zK-IieDEqo1l/view
> 
> Also, I'm seeing the step indexes in the flow artifacts (the on-disk files) are 1-based, not 0-based.

@wooldridge - Fixed these issues and committed the changes. Please review now./

srinathgit commented:
@xnikhil08 ,
If no value is set for an option, we don't have to include that in the mlcp command. For example, Target URI Replacement is not set in this case and we don't have to include -output_uri_replace option in the mlcp command

![image](https://user-images.githubusercontent.com/5034960/60128380-5c636200-9747-11e9-8b35-be54c635c95a.png)

srinathgit commented on web/src/main/ui/app/components/flows-new/edit-flow/ingest/ui/ingest-ui.component.ts:
```diff
@@ -127,7 +151,82 @@ export class IngestUiComponent implements OnInit{
+    const username = this.envService.settings.mlUsername;
+    let input_file_path = this.step.fileLocations.inputFilePath;
+    let input_file_type = this.step.fileLocations.inputFileType;
+    let document_type = this.step.options.outputFormat.toLowerCase();
```
The value for input_file_type for mlcp is not the same as the one chosen from the dropdown. Please check the link https://docs.marklogic.com/guide/mlcp/import#id_71112

xnikhil08 replied:
This is Fixed. Please review now.

xnikhil08 commented:
> @xnikhil08 when I create a new ingestion step in my flow and select it, I'm seeing step=0 in the MLCP command for that step:
> 
> https://drive.google.com/file/d/1twVcX4zo2Na0cRq07ieVCIn7DZ2Qmxou/view
> 
> In the above case, it should read: step=2

@wooldridge :- Implemented the changes and fixed all the issues. Please review now.
